grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1141,4 +1142,46 @@ class ProductService {
         }
         return results
     }
+
+    def searchProductDtos(String[] terms) {
+        def query = """
+select p.id, p.name, p.product_code as productCode, p.cold_chain as coldChain, p.controlled_substance as controlledSubstance, p.hazardous_material as hazardousMaterial, p.reconditioned,
+ (select max(pc.color) from product_catalog_item pci left outer join product_catalog pc on pci.product_catalog_id = pc.id where pci.product_id = p.id group by pci.product_id) as productColor
+ from product p
+ left outer join product_supplier productsup1_ on p.id=productsup1_.product_id
+ left outer join party manufactur2_ on productsup1_.manufacturer_id=manufactur2_.id and (${terms.collect { "lower(manufactur2_.name) like '%${it}%'" }.join(" or ")})
+ left outer join party supplier_a3_ on productsup1_.supplier_id=supplier_a3_.id and (${terms.collect { "lower(supplier_a3_.name) like '%${it}%'" }.join(" or ")})
+ left outer join inventory_item inventoryi4_ on p.id=inventoryi4_.product_id and (${terms.collect { "lower(inventoryi4_.lot_number) like '%${it}%'" }.join(" or ")})
+ 
+ where p.active=1 and
+ (${terms.collect { """lower(p.name) like '%${it}%' or lower(p.product_code) like '%${it}%' or lower(p.description) like '%${it}%'
+ or lower(p.brand_name) like '%${it}%' or lower(p.manufacturer) like '%${it}%' or lower(p.manufacturer_code) like '%${it}%'
+ or lower(p.manufacturer_name) like '%${it}%' or lower(p.vendor) like '%${it}%' or lower(p.vendor_code) like '%${it}%'
+ or lower(p.vendor_name) like '%${it}%' or lower(p.upc) like '%${it}%' or lower(p.ndc) like '%${it}%'
+ or lower(p.unit_of_measure) like '%${it}%'
+ 
+ or lower(productsup1_.name) like '%${it}%' or lower(productsup1_.code) like '%${it}%'
+ or lower(productsup1_.product_code) like '%${it}%' or lower(productsup1_.manufacturer_code) like '%${it}%'
+ or lower(productsup1_.manufacturer_name) like '%${it}%' or lower(productsup1_.supplier_code) like '%${it}%'
+ or lower(productsup1_.supplier_name) like '%${it}%'""" }.join(" or ")}
+ 
+ or manufactur2_.id is not null
+ or supplier_a3_.id is not null
+ or inventoryi4_.id is not null)
+
+group by p.id
```

Comment:
Is this allowed? MySQL usually complains if you include all of the columns. Instead of a group by you could probably do a distinct before p.id in the SELECT and that should give us unique results.